include/shark/Data/SparseData.h

```diff
@@ -157,20 +197,109 @@ void exportSparseData(LabeledData<InputType, unsigned int>& dataset, const std::
 		if(sortLabels) i = L[ii].second;
 		else i = ii;
 		// apply transformation to label and write it to file
-		if(oneMinusOne) ofs << 2*int(dataset.element(i).label)-1 << " ";
+		if(oneMinusOne) stream << 2*int(dataset.element(i).label)-1 << " ";
 		//libsvm file format documentation is scarce, but by convention the first class seems to be 1..
-		else ofs << dataset.element(i).label+1 << " ";
+		else stream << dataset.element(i).label+1 << " ";
 		// write input data to file
 		for(std::size_t j=0; j<dim; j++) {
 			if(dense) 
-				ofs << " " << j+1 << ":" <<dataset.element(i).input(j);
+				stream << " " << j+1 << ":" <<dataset.element(i).input(j);
 			else if(dataset.element(i).input(j) != 0) 
-				ofs << " " << j+1 << ":" << dataset.element(i).input(j);
+				stream << " " << j+1 << ":" << dataset.element(i).input(j);
+		}
+		stream << std::endl;
+	}
+}
+
+/// \brief Export classification data to sparse data (libSVM) format.
+///
+/// \param  dataset     Container storing the data
+/// \param  fn          Output file name
+/// \param  dense       Flag for writing all components, including zeros
```

Comment:
> why would i want to write a dense dataset anyways? :/

VERY good question. I just tried to keep compatibility with the existing 
interface, but my first reflex was: remove!